fitstack/utils.py

```diff
@@ -271,6 +272,89 @@ def initialize_pol(cnt, pol=None, combine=False):
     return stack, weight, cpol
 
 
+def average_stacks(stacks, pol=None, combine=True, sort=True):
+    """Calculate the mean and variance of a set of stacks.
+
+    Parameters
+    ----------
+    stack : MockFrequencyStackByPol
+        Set of stacks to average.
+    pol : list of str
+        The polarisations to select.  If not provided,
+        then ["XX", "YY"] is assumed.
+    combine : bool
+        Add an element to the polarisation axis that is
+        the weighted sum of XX and YY.  Default is True.
+    sort : bool
+        Sort the frequency offset axis in ascending order.
+        Default is True.
+
+    Returns
+    -------
+    avg : FrequencyStackByPol
+        Container that has collapsed over the mock axis.
+        The stack dataset contains the mean and the weight
+        dataset contains the inverse variance.
+    """
+
+    sarr, _, spol = initialize_pol(stacks, pol=pol, combine=combine)
+    nstack = sarr.shape[0]
+
+    freq = stacks.freq
+    if sort:
+        isort = np.argsort(freq)
+        freq = freq[isort]
+        sarr = sarr[..., isort]
+
+    avg = containers.FrequencyStackByPol(
+        pol=np.array(spol), freq=freq, attrs_from=stacks
+    )
+
+    avg.attrs["num"] = nstack
+    avg.stack[:] = np.mean(sarr, axis=0)
+    avg.weight[:] = tools.invert_no_zero(np.var(sarr, axis=0))
+
+    return avg
+
+
+def load_pol(filename, pol=None):
```

Comment:
We should really fix this in `ContainerBase` at some point. I didn't think it would be so bad, but it's been super annoying recently.

Reply:
Right I think the issue that I was having is that I could not detect the class _and_ perform an axis selection.  I believe it was using the axis specification for whatever class I used to call `from_file`, not the actual class that was saved to disk and indicated in the `__memh5_subclass` attribute.  This came about when trying to load both `FrequencyStackByPol` and `MockFrequencyStackByPol` containers and perform a polarization selection.